internal/metrics/metrics.go

```diff
@@ -0,0 +1,125 @@
+package metrics
+
+import (
+	"context"
+	"os"
+	"runtime"
+	"time"
+
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/mem"
+	"github.com/shirou/gopsutil/net"
+	"github.com/shirou/gopsutil/process"
+)
+
+type Metrics struct {
+	ProcessCPU       ProcessCPU
+	CPU              map[string]CPU
+	NIC              map[string]NIC
+	Memory           Memory
+	CPUPercent       float64
+	GarbageCollector GarbageCollector
+}
+
+type GarbageCollector struct {
+	NumGC uint64
+}
+
+type ProcessCPU struct {
+	User   float64
+	System float64
+}
+
+type CPU struct {
+	User   float64
+	System float64
+	Usage  float64
+	Total  float64
```

Comment:
I think originally there was also interest in capturing nice and/or steal. Is this no longer needed?

(Also can  definitely be added later!)

Reply:
Correct, these are now being summed as part of usage rather than individually:
https://github.com/lightstep/lightstep-tracer-go/pull/249/files/a847ca7d4d70b7b8b5a798d42c6bc6e3554d22f5#diff-501d4915da42d43fcfa855ffcc089cfeR104